packages/app/src/components/SourceForm.tsx

```diff
@@ -641,10 +646,39 @@ export function MetricTableModelForm({
   setValue: UseFormSetValue<TSource>;
 }) {
   const databaseName = watch(`from.databaseName`, DEFAULT_DATABASE);
-  const tableName = watch(`from.tableName`);
   const connectionId = watch(`connection`);
 
-  const [showOptionalFields, setShowOptionalFields] = useState(false);
+  useEffect(() => {
+    setValue('timestampValueExpression', '');
+    const { unsubscribe } = watch(async (value, { name, type }) => {
+      try {
+        if (name && type === 'change') {
+          const [prefix, suffix] = name.split('.');
+          if (prefix === 'metricTables') {
+            const tableName =
+              value?.metricTables?.[suffix as keyof typeof value.metricTables];
+            const metricType = suffix as MetricsDataType;
+            const isValid = await isValidMetricTable({
+              databaseName,
+              tableName,
+              connectionId,
+              metricType,
+            });
+            if (!isValid) {
+              notifications.show({
+                color: 'red',
+                message: `${tableName} is not a valid OTEL ${metricType} schema.`,
+              });
+            }
+          }
+        }
+      } catch (e) {
+        console.error(e);
```

Comment:
nit: Should probably toast here with `e.message`

Reply:
Added just below but left in the `console.error` so we also output more context for debugging later.